Microsoft.Toolkit.Uwp.UI.Animations/Behaviors/ViewportBehavior.cs

```diff
@@ -0,0 +1,174 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Animations.Behaviors
+{
+    /// <summary>
+    /// A class for listening element enter or leave the viewport
+    /// </summary>
+    public class ViewportBehavior : BehaviorBase<FrameworkElement>
+    {
+        /// <summary>
+        /// The IsFullyInViewport value of the associated element
+        /// </summary>
+        public static readonly DependencyProperty IsFullyInViewportProperty =
+            DependencyProperty.Register(nameof(IsFullyInViewport), typeof(bool), typeof(ViewportBehavior), new PropertyMetadata(default(bool), OnIsFullyInViewportChanged));
+
+        /// <summary>
+        /// The IsInViewport value of the associated element
+        /// </summary>
+        public static readonly DependencyProperty IsInViewportProperty =
+            DependencyProperty.Register(nameof(IsInViewport), typeof(bool), typeof(ViewportBehavior), new PropertyMetadata(default(bool), OnIsInViewportChanged));
+
+        /// <summary>
+        /// Associated element fully enter the viewport event
+        /// </summary>
+        public event EventHandler EnteredViewport;
+
+        /// <summary>
+        /// Associated element fully leave the viewport event
+        /// </summary>
+        public event EventHandler LeavedViewport;
```

Comment:
This should be ExitedViewport.

Reply:
@JustinXinLiu Done. Changed all "leave" words to "exit".